tests/src/reflection/GenericAttribute/GenericAttributeTests.cs

```diff
@@ -0,0 +1,163 @@
+        AssertAny(a3, a => (a as MultiAttribute<bool>)?.Value == true);
+
+        var b3 = ((ICustomAttributeProvider)deriveTypeInfo).GetCustomAttributes(true);
+        AssertAny(b3, a => a is SingleAttribute<int>);
+        AssertAny(b3, a => a is SingleAttribute<bool>);
+        AssertAny(b3, a => (a as MultiAttribute<int>)?.Value == 0);
+        AssertAny(b3, a => (a as MultiAttribute<int>)?.Value == 1);
+        AssertAny(b3, a => (a as MultiAttribute<int>)?.Value == 2);
+        AssertAny(b3, a => (a as MultiAttribute<bool>)?.Value == false);
+        AssertAny(b3, a => (a as MultiAttribute<bool>)?.Value == true);
+
+        var a4 = CustomAttributeExtensions.GetCustomAttributes<SingleAttribute<int>>(programTypeInfo, true);
+        AssertAny(a4, a => a is SingleAttribute<int>);
+
+        var b4 = ((ICustomAttributeProvider)programTypeInfo).GetCustomAttributes(typeof(SingleAttribute<int>), true);
+        AssertAny(b4, a => a is SingleAttribute<int>);
+
+        var a5 = CustomAttributeExtensions.GetCustomAttributes<SingleAttribute<bool>>(programTypeInfo);
+        AssertAny(a5, a => a is SingleAttribute<bool>);
+
+        var b5 = ((ICustomAttributeProvider)programTypeInfo).GetCustomAttributes(typeof(SingleAttribute<bool>), true);
+        AssertAny(b5, a => a is SingleAttribute<bool>);
+
+        var a6 = CustomAttributeExtensions.GetCustomAttributes<MultiAttribute<int>>(programTypeInfo, true);
+        AssertAny(a6, a => (a as MultiAttribute<int>)?.Value == 0);
+        AssertAny(a6, a => (a as MultiAttribute<int>)?.Value == 1);
+        AssertAny(a6, a => (a as MultiAttribute<int>)?.Value == 2);
+
+        var b6 = ((ICustomAttributeProvider)programTypeInfo).GetCustomAttributes(typeof(MultiAttribute<int>), true);
+        AssertAny(b6, a => (a as MultiAttribute<int>)?.Value == 0);
+        AssertAny(b6, a => (a as MultiAttribute<int>)?.Value == 1);
+        AssertAny(b6, a => (a as MultiAttribute<int>)?.Value == 2);
+
+        var a7 = CustomAttributeExtensions.GetCustomAttributes<MultiAttribute<bool>>(programTypeInfo, true);
+        AssertAny(a7, a => (a as MultiAttribute<bool>)?.Value == false);
+        AssertAny(a7, a => (a as MultiAttribute<bool>)?.Value == true);
+
+        var b7 = ((ICustomAttributeProvider)programTypeInfo).GetCustomAttributes(typeof(MultiAttribute<bool>), true);
+        AssertAny(b7, a => (a as MultiAttribute<bool>)?.Value == false);
+        AssertAny(b7, a => (a as MultiAttribute<bool>)?.Value == true);
+
+        var a8 = CustomAttributeExtensions.GetCustomAttributes<MultiAttribute<bool?>>(programTypeInfo, true);
+        AssertAny(a8, a => (a as MultiAttribute<bool?>)?.Value == null);
+
+        var b8 = ((ICustomAttributeProvider)programTypeInfo).GetCustomAttributes(typeof(MultiAttribute<bool?>), true);
+        AssertAny(b8, a => (a as MultiAttribute<bool?>)?.Value == null);
+
+        Assert(CustomAttributeExtensions.GetCustomAttributes(programTypeInfo, typeof(MultiAttribute<>), false) == null);
```

Comment:
I don't think it's reasonable to expect these last three tests to work. These apis are statically declared to return `Attribute` arrays but they actually return arrays of the specific attribute type, i.e. this works:

`     MyAttribute[] a = (MyAttribute[])(CAE.GetCustomAttributes(t, typeof(MyAttribute));
`

But you can't instantiate an array of an open type like `MyAttribute<>` so I would expect this to throw with an exception from `Array.Instance`.